### PR TITLE
CAPSTONE GROUP PROJECT

### PR DESCRIPTION
**Pull request by Josphat205**
## Team
@Shubh-Dev 
@ibrahim777764 
@Josphat205 
## Procedures

- Implemented classes and objects in Ruby
- Implemented encapsulation and inheritance with Ruby
- Set up associations between classes and objects
- Stored data in json file.
- Created SQL queries 

# Presentation video link

- [Link](https://drive.google.com/file/d/1Th8hLPe8O6h_7KCHoD7xVDpi0hidgkKH/view?usp=sharing)

### CODE REVIEW

File: music/genres.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    @id = rand(1..1000)
+    @name = name
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+    item.add_genre = self
+  end
+end
```
Comment (KDlamini) [inline music/genres.rb:16]: There's no option for creating a new Genre on the menu. When I list Genres I see album names. The Genre class is its own entity therefore it should have its own data that is separate from the album and at the same time associated with the album.

image

- You can add an option to create a new genre or choose from existing genres when creating a new album. Then list all genres from the Genre class.
- I should be able to list or create something like Hip Hop, Pop, Rock, RnB, Soul, Jazz, etc. from the Genres class.

image

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  puts 'List of things'
+  puts '========================'
+  puts '0 Save and Exit'
+  puts '1 List all books'
+  puts '2 List all music albums'
+  puts '3 List all games'
+  puts '4 List all genres'
+  puts '5 List all Labels'
+  puts '6 List all authors'
+  puts '7 List all sources'
```
Comment (KDlamini) [inline main.rb:31]: Since you are not doing `sources` in this group, it is unnecessary to have the sources option. Kindly remove it.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @Josphat205, @ibrahim777764, @Shubh-Dev 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working basic UI for Catalogue of My Things according to the project requirements. 🎊
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ugboaja-Uchechi — APPROVED** [review]
Hi @ibrahim777764,

## Status: APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 
- Great job with your design and functionalities 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**